wiki/Bazel-User-Guide.md

```diff
@@ -0,0 +1,277 @@
+     or
+   * Right-click on the file tab and then click `copy path/reference`.
+   * Next, select `copy path from repository root`. 
+   * Paste the copied path to the bazel test command in the terminal and remove the `.kt` extension, e.g. `bazel test domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest`.
+
+With this syntax, Bazel implicitly converts the file path into a target if it matches a test rule in the BUILD file. If the file is not part of a test target in the BUILD file, this command may fail.
+**When to use this**: If you are unsure about the exact Bazel target and want a quick way to run a test.
+
+3. Querying for a target
+Using the explicit Bazel target (// syntax) is more reliable compared to the file path syntax above. However, in cases such as running the test outside android studio, you may not know the target. In this case, you can use Bazel's `query` command to retrieve the correct test target:
+   * Copy the relative path of the test file.
+   * Run the following command in your terminal to get the test’s Bazel target: `bazel query relative-path-of-test-file`
+   Example
+   
+   ```shell
+   bazel query domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest.kt
+   ```
+   * The output will be a Bazel target that starts with //. Copy the target and remove the .kt extension.
+   * Use this target in the bazel test command, e.g. `bazel test //domain/src/test/java/org/oppia/android/domain/onboarding:AppStartupStateControllerTest`.
+
+### Common Flags
+
+`--test_output=all` → Show full test output.
+`--cache_test_results=no` → Force re-running tests even if cached.
+`--test_filter=ClassName#methodName` → Run specific test cases.
+
+### Running multiple test targets
+To run all the test targets in the app module:
+
+```
+bazel test //app/...
+```
+
+To run all the test targets in the project (note that this would be extremely slow and is not recommended):
+
+```
+bazel test //...
+```
+
+## Concepts and Terminology
+**[Workspace](https://github.com/oppia/oppia-android/blob/develop/WORKSPACE)**
+
+A workspace is a directory where we specify the targeted SDK version, required dependencies, and corresponding Rules. The directory containing the WORKSPACE file is the root of the main repository, which in our case, is the `oppia-android` root directory.
+**[Packages](https://github.com/oppia/oppia-android/tree/develop/app)**
+
+A package is defined as a directory containing a file named BUILD or BUILD.bazel.
+
+**[Binary rules](https://github.com/oppia/oppia-android/blob/ba8d914480251e4a8543feb63a93b6c91e0a5a2f/BUILD.bazel#L3)**
+
+A rule specifies the relationship between inputs and outputs, and the steps to build the outputs.
+In Android, rules are defined using `android_binary`. Android rules for testing are `android_instrumentation_test` and `android_local_test`.
+
+**[BUILD files](https://github.com/oppia/oppia-android/blob/develop/app/BUILD.bazel)**
```

Comment:
Please permalink to this & the WORKSPACE files, too. Even these files aren't permanent and will probably disappear one day (app's eventually with modularization maturity and WORKSPACE when we migrate to Bazel 8.x).

Reply:
Done